gpkit/constraints/relax.py

```diff
@@ -0,0 +1,140 @@
+"""Models for assessing primal feasibility"""
+from .model import Model
+from .set import ConstraintSet
+from ..nomials import Variable, VectorVariable, parse_subs, NomialArray
+
+
+class RelaxAll(Model):
+    """Relax constraints the same amount, as in Eqn. 10 of [Boyd2007].
+
+    Arguments
+    ---------
+    constraints : iterable
+        Constraints which will be relaxed (made easier).
+
+    varname : str
+        LaTeX name of relaxation variable.
+
+
+    Attributes
+    ----------
+    relaxvar : Variable
+        The variable controlling the relaxation. A solved value of 1 means no
+        relaxation. Higher values indicate the amount by which all constraints
+        have been made easier: e.g., a value of 1.5 means all constraints were
+        50 percent easier in the final solution than in the original problem.
+
+    [Boyd2007] : "A tutorial on geometric programming", Optim Eng 8:67-122
+
+    """
+
+    def __init__(self, constraints):
+        if not isinstance(constraints, ConstraintSet):
+            constraints = ConstraintSet(constraints)
+        substitutions = dict(constraints.substitutions)
+        posynomials = constraints.as_posyslt1()
+        self.relaxvar = Variable("C", models=["Relax"],
+                                 modelnums=[Model._nums["Relax"]])
+        Model.__init__(self, self.relaxvar,
+                       [[posy <= self.relaxvar
+                         for posy in posynomials],
+                        self.relaxvar >= 1],
+                       substitutions,
+                       name=False)
```

Comment:
or perhaps RelaxConstraints should be capable of producing this formulation as a special case, e.g. by a boolean keyword arg?

The gp_tutorial lists two different versions of the RelaxConstraints formulation (equations 10 and 11).